### ISSUE TITLE
Change the way we stitch MIME parts together for display

### ISSUE THREAD

**Issue by cketti** [k9mail/k-9#3388]
Right now, if a message contains multiple displayable HTML parts, we first concatenate all of them and only then process the resulting HTML (remove JavaScript etc). See https://github.com/k9mail/k-9/blob/e680c84085d82a0c1d0b06f10bf66ff7f42df907/k9mail/src/main/java/com/fsck/k9/mailstore/MessageViewInfoExtractor.java#L186-L257

This approach has lead to several clients being vulnerable to what is described as the Direct Exfiltration attack at <https://efail.de/>. Due to the way we handle encrypted content we're not vulnerable to EFAIL. But it's still very undesirable that the contents of a preceding MIME part can influence the way the following HTML part is parsed. So we should fix that by processing the different MIME parts individually and then stitching together the processed outputs.

**Comment by benbucksch** [comment]
> if a message contains multiple displayable HTML parts, we first concatenate all of them and only then process the resulting HTML (remove JavaScript etc)

Right. That's what Thunderbird does as well, and what leads to the security bug. So, we plan to run each MIME part (if it's HTML) through the HTML parser separately.

**Comment by DPTJKKVH** [comment]
Could that lead to a plaintext only display option as well?

**Comment by Valodim** [comment]
Note that this is not relevant for pgp, because we *don't* stitch decrypted parts and plaintext ones.
